hypershift-operator/controllers/nodepool/azure.go

```diff
@@ -16,18 +16,7 @@ import (
 	capiazure "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 )
 
-func azureMachineTemplateSpec(hcluster *hyperv1.HostedCluster, nodePool *hyperv1.NodePool, existing capiazure.AzureMachineTemplateSpec) (*capiazure.AzureMachineTemplateSpec, error) {
```

Comment:
why is this PR dropping this?

Reply:
Just dropping the sshKey part out of this function.

Reply:
I thought maybe we could drop the sshKey piece; not sure if we need this now. When I tried that, there is an upstream error where this field is not omitted and is initialized as "" which is invalid.

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5092